### PR TITLE
refactor: Improve scaffolding

### PR DESCRIPTION
**Pull request by paurakhsharma**

### CODE REVIEW

**Review by JannieT — CHANGES_REQUESTED** [review]
Can you please:
- try the suggestion to avoid watching the widget tree root
- add a network service with a good body perser like we have in [Alpha Plus app](https://dev.azure.com/alphaglobal/GuestExperience/_git/GuestExperienceMobile?path=/lib/src/shared/services/network_service.dart)

File: flutter/lib/src/app.dart [hunk 1]
```diff
@@ -31,7 +33,7 @@ class MyApp extends StatelessWidget {
           // app theme
           theme: styles.themeLight,
           darkTheme: styles.themeDark,
-          themeMode: get<SettingsManager>().themeMode,
+          themeMode: themeMode,
```
[the remark this answers is not in the document]
Comment (paurakhsharma) [reply flutter/lib/src/app.dart:36]: Previous approach was doing the same thing as well. We should rebuild the whole app when the theme is changed. I think the solution you suggested would do the same thing as well. But I will experiment to see if that is correct. Thank you for the suggestion.

**Comment by paurakhsharma** [comment]
@JannieT PR updated.

File: flutter/init.js [hunk 2]
```diff
@@ -118,7 +122,7 @@ const createApp = async () => {
     { default: 'test_app', noClear: true },
   );
 
-  const defaultBundleId = `co.scoutredeem.${appName}`;
+  const defaultBundleId = `co.scoutredeem`;
```
[the remark this answers is not in the document]
Comment (paurakhsharma) [reply flutter/init.js:124]: No the last one was incorrect, the org value should not include the package name.

**Pull request merged**